### PR TITLE
block double clicks on download

### PR DESCRIPTION
**Pull request by benrudolph**
@orangejenny not the most angular way to do things, but didn't want to mutate the export model or do some other more invasive fix. this stops users from spamming the download button

http://manage.dimagi.com/default.asp?231800

cc: @calellowitz @sravfeyn

### CODE REVIEW

**Comment by benrudolph** [comment]
screen shot 2016-07-12 at 3 52 31 pm

**Comment by orangejenny** [comment]
I kind of wish this extended the `.disable-on-submit` code in hq.helpers rather than being a one-off, but understand if it wasn't worth getting into for fixing this one area.

Does the button ever get re-enabled? Or does the download take you to another page?

File: corehq/apps/export/static/export/js/list_exports.ng.js [hunk 1]
```diff
@@ -80,6 +80,11 @@
             var currentUrl = useLegacyBulkExportUrl ? $scope.legacy_bulk_download_url : $scope.bulk_download_url;
             input.closest("form").attr("action", currentUrl);
         };
+        $scope.downloadRequested = function($event) {
```
Comment (orangejenny) [inline corehq/apps/export/static/export/js/list_exports.ng.js]: Is the `$` prefix an angular convention? It made me think `$event` was a jQuery object for a second.

File: corehq/apps/export/static/export/js/list_exports.ng.js [hunk 2]
```diff
@@ -80,6 +80,11 @@
...
+            var $btn = $($event.target);
+            $btn.addClass('disabled');
```
Comment (orangejenny) [inline corehq/apps/export/static/export/js/list_exports.ng.js]: Think you also need to set the `disabled` attribute.

Comment (orangejenny) [inline corehq/apps/export/static/export/js/list_exports.ng.js]: Oh, it's a link, no `disabled` attribute, need to...switch the `href`, I suppose.

File: corehq/apps/export/static/export/js/list_exports.ng.js (hunk 1 above)
Comment (benrudolph) [inline corehq/apps/export/static/export/js/list_exports.ng.js]: angular convention 😕

File: corehq/apps/export/static/export/js/list_exports.ng.js (hunk 2 above)
Comment (benrudolph) [inline corehq/apps/export/static/export/js/list_exports.ng.js]: i don't think you need to switch the href in this case. the `disabled` class adds `pointer-events: none;` which means they can't click it

**Comment by benrudolph** [comment]
the button only gets re enabled on refresh since it's a daily saved export, i don't think there's a lot of use case to downloading multiple of the same thing. if they do it's probably an accident

**Comment by benrudolph** [comment]
the reason why `disable-on-submit` doesn't work is this doesn't use any click handle, the browser just interprets the response and knows not to switch pages (i believe)

File: corehq/apps/export/static/export/js/list_exports.ng.js (hunk 2 above)
Comment (orangejenny) [inline corehq/apps/export/static/export/js/list_exports.ng.js]: Aha. I hadn't heard of `pointer-events` (though it looks like it's been around for a long time). Nice.

**Comment by orangejenny** [comment]
Cool, that all makes sense. Good to merge on tests passing.

File: corehq/apps/export/static/export/js/list_exports.ng.js (hunk 2 above)
Comment (benrudolph) [inline corehq/apps/export/static/export/js/list_exports.ng.js]: it's my claim to fame on stackoverflow: http://stackoverflow.com/questions/10867506/dragleave-of-parent-element-fires-when-dragging-over-children-elements/14027995#14027995 😄

Comment (orangejenny) [inline corehq/apps/export/static/export/js/list_exports.ng.js]: ha, high-quality SO answer there

**Pull request merged**